### PR TITLE
Laboratory work #3, Tormashova Valeriya - 19FPL2

### PR DESCRIPTION
**Pull request by LeraTormashova**

### CODE REVIEW

File: lab_3/main.py [hunk 1]
```diff
@@ -17,38 +19,65 @@ def tokenize_by_sentence(text: str) -> tuple:
...
         """
         Puts a letter into storage, assigns a unique id
         :param letter: a letter
         :return: 0 if succeeds, 1 if not
         """
-        pass
+        if not isinstance(letter, str) or not letter:
+            return 1
+
+        id = len(self.storage)
```
Comment (ValeriyaKuznetsova) [inline lab_3/main.py]: can we not count this len every time?

File: lab_3/main.py [hunk 2]
```diff
@@ -59,42 +88,99 @@ def encode_corpus(storage: LetterStorage, corpus: tuple) -> tuple:
...
     """
-    pass
+    if not isinstance(corpus, tuple) or not corpus or not isinstance(storage, LetterStorage):
+        return ()
+
+    new_corpus = []
+
+    for sentence in corpus:
+        if not isinstance(sentence[0], tuple):
+            [storage._put_letter(letter) for letter in sentence]
```
Comment (ValeriyaKuznetsova) [inline lab_3/main.py]: a bit hard-to-read line

File: lab_3/main.py [hunk 3]
```diff
@@ -59,42 +88,99 @@ def encode_corpus(storage: LetterStorage, corpus: tuple) -> tuple:
...
+            new_corpus.append(tuple(tuple([storage.get_id_by_letter(letter)]) for letter in sentence))
+        else:
+            for word in sentence:
+                [storage._put_letter(letter) for letter in word]
```
Comment (ValeriyaKuznetsova) [inline lab_3/main.py]: same

File: lab_3/main.py [hunk 4]
```diff
@@ -59,42 +88,99 @@ def encode_corpus(storage: LetterStorage, corpus: tuple) -> tuple:
...
     :return: a tuple of the encoded sentences
     """
-    pass
+    if not isinstance(corpus, tuple) or not corpus or not isinstance(storage, LetterStorage):
+        return ()
+
+    new_corpus = []
+
+    for sentence in corpus:
+        if not isinstance(sentence[0], tuple):
```
Comment (ValeriyaKuznetsova) [inline lab_3/main.py]: can it be not tuple?

**Comment by ValeriyaKuznetsova** [comment]
don't forget about target score if you don't want 10

and start.py

**Review by ValeriyaKuznetsova — COMMENTED** [review]
change start.py and make it green

everything seems OK

**Review by ValeriyaKuznetsova — COMMENTED** [review]
Why 6 if you have completed the functions for 8?

**Comment by LeraTormashova** [comment]
But I haven't added realisation of the last function for score 8 and not sure that I'll do it later

**Pull request closed without merging**